api/src/api/services/phases/handlers/brla-teleport-handler.ts

```diff
@@ -57,25 +62,42 @@ export class BrlaTeleportPhaseHandler extends BasePhaseHandler {
         moonbeamEphemeralAddress,
         inputAmountBeforeSwapRaw, // TODO verify this is okay, regarding decimals.
         pollingTimeMs,
-        maxWaitingTimeMs,
+        EVM_BALANCE_CHECK_TIMEOUT_MS,
         moonbeam,
       );
 
       // Add delay to ensure the transaction is settled
       await new Promise((resolve) => setTimeout(resolve, 12000)); // 12 seconds, 2 moonbeam blocks.
+
     } catch (balanceCheckError) {
-      if (balanceCheckError instanceof Error) {
-        if (balanceCheckError.message === 'Balance did not meet the limit within the specified time') {
-          throw new Error(`BrlaTeleportPhaseHandler: balanceCheckError ${balanceCheckError.message}`);
-        } else {
-          logger.error('Error checking Moonbeam balance:', balanceCheckError);
-          throw new Error(`BrlaTeleportPhaseHandler: Error checking Moonbeam balance`);
-        }
+      if (!(balanceCheckError instanceof Error)) throw balanceCheckError;
+      const { message } = balanceCheckError;
+    
+      const isCheckTimeout = message.includes('did not meet the limit within the specified time');
+      if (isCheckTimeout && this.finalPaymentTimeout(state)) {
+        logger.error('Payment timeout:', balanceCheckError);
+        return this.transitionToNextPhase(state, 'failed');
       }
+    
+      throw isCheckTimeout
+        ? this.createRecoverableError(`BrlaTeleportPhaseHandler: ${message}`)
+        : new Error(`Error checking Moonbeam balance: ${message}`);
     }
 
     return this.transitionToNextPhase(state, 'moonbeamToPendulumXcm');
   }
+
+  protected finalPaymentTimeout(state: RampState): boolean {
```

Comment:
I think the name is not great. It returns a `boolean` so maybe we make it start with `is`. How about
```suggestion
  protected isPaymentTimeoutReached(state: RampState): boolean {
```